scripts/helm-readmes.sh

```diff
@@ -0,0 +1,176 @@
+      echo
+      exit
+    fi
+  done
+}
+
+verify_readmes() {
+  local READMES=("${AGENT}" "${AGGREGATOR}" "${UNIFIED}")
+  for i in "${READMES[@]}"; do
+    if [[ -f "${i}" ]]; then
+      echo "*********************************"
+      echo "Previous version exists at ${i}"
+      echo " Please rerun with 'update' subcommand"
+      echo "*********************************"
+      echo
+      exit 1
+    fi
+  done
+}
+
+remove_old() {
+  local READMES=("${AGENT}" "${AGGREGATOR}" "${UNIFIED}")
+  for i in "${READMES[@]}"; do
+    if [[ -f "${i}" ]]; then
+      echo "*********************************"
+      echo "Removing old version of ${i}"
+      echo "*********************************"
+      echo
+      rm "${i}"
+    fi
+  done
+}
+
+clean_tmpfiles() {
+  if [[ -f ${ROOT_DATA} ]]; then
+    echo "*********************************"
+    echo "Existing tmpfile detected. Removing ${ROOT_DATA}"
+    echo "*********************************"
+    echo
+    rm ${ROOT_DATA}
+  fi
+}
+
+generate_readme() {
+  clean_tmpfiles
+  echo "*********************************"
+  echo "Generating tmpfile from cue data"
+  echo "*********************************"
+  echo
+
+  ./scripts/cue.sh export ./docs/**/*.cue | jq -r '.installation.operating_systems.ubuntu.interfaces[] | select(.title == "Helm 3")'>${ROOT_DATA}
+
+  local DESCRIPTION=$(cat "${ROOT_DATA}" | jq -r .roles.${1}.description | tr '\n' ' ')
+  local ADD_REPO=$(cat "${ROOT_DATA}" | jq -r .roles.${1}.commands.add_repo)
+  local CHECK_OPTS=$(cat "${ROOT_DATA}" | jq -r .roles.${1}.commands.helm_values_show)
+  local VECTOR_CONFIG=$(cat "${ROOT_DATA}" | jq -r .roles.${1}.commands.configure | awk '{gsub(/\\n/,"\n")}1')
+  local VECTOR_INSTALL=$(cat "${ROOT_DATA}" | jq -r .roles.${1}.commands.install | awk '{gsub(/\\n/,"\n")}1')
+
+  if [ "${1}" == "agent" ]; then
+     local README="${AGENT}"
+  elif [ "${1}" == "aggregator" ]; then
+    local README="${AGGREGATOR}"
+  elif [ "${1}" == "unified" ]; then
+    local README="${UNIFIED}"
+  fi
+
+  echo "*********************************"
+  echo "Writing Agent README.md"
+  echo "*********************************"
+  echo
+  sh -c "cat > ${README}" <<EOT
+# [Vector](https://vector.dev) Helm Chart
+
+This is an opinionated Helm Chart for running [Vector](https://vector.dev) in Kubernetes.
+
+Our charts use Helm's dependency system, however we only use local dependencies.
+Head over to the repo for [more information on development and contribution](https://github.com/timberio/vector/tree/master/distribution/helm).
+
+${DESCRIPTION}
+
+To get started add the Helm chart repo
+\`\`\`
+${ADD_REPO}
+\`\`\`
+
+Check the available Helm chart configuration options
+\`\`\`
+${CHECK_OPTS}
+\`\`\`
+
+Set up a Vector config  that leverages our \`kubernetes_logs\` data source
+\`\`\`
+${VECTOR_CONFIG}
+\`\`\`
+
+To install the chart
+\`\`\`
+${VECTOR_INSTALL}
+\`\`\`
+
+To update
+\`\`\`
+${VECTOR_INSTALL}
+\`\`\`
+EOT
```

Comment:
What if we move this whole section into some cue file as well? Or extract into a template file - but since we're using cue - might make sense to keep it all there.

Reply:
Most of the bulk of this data is coming out of cue already. I'm not totally certain what nesting looks like in cue but I could research. Or perhaps @lucperkins has an idea. Ulitmately it would be a new cue entry that includes data templated from other cue data?

Reply:
Yeah, that's the idea. I don't see why not... cc @binarylogic

Reply:
Having spent the afternoon and morning looking into this I'm feeling like further templating of this information is not a good use of time. 

Ultimately the text here is a hair shy of being static content. Everything else, everything that might have a common value change is already templated out so adding some significant overhead to increase layers of templating for a handful of lines that will never be rendered outside of this singular file feels like kind of a waste.

If there are strong enough opinions then I'll cede this but right now I'm very disinclined to spend any more time on this.